x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -32,55 +41,76 @@ export class MBMapContainer extends React.Component {
     }
   }, 256);
 
-  _updateTooltipState = _.debounce(async (e) => {
 
-    const mbLayerIds = this._getMbLayerIdsForTooltips();
-    const features = this._mbMap.queryRenderedFeatures(e.point, { layers: mbLayerIds });
+  _lockTooltip = async (e) => {
 
+    const features = this._getFeaturesUnderPointer(e.point);
     if (!features.length) {
       this.props.setTooltipState(null);
       return;
     }
 
     const targetFeature = features[0];
+    const layer = this._getLayer(targetFeature.layer.id);
+    const popupAnchorLocation = this._justifyAnchorLocation(e.lngLat, targetFeature);
+    this.props.setTooltipState({
+      type: TOOLTIP_TYPE.LOCKED,
+      layerId: layer.getId(),
+      featureId: targetFeature.properties[FEATURE_ID_PROPERTY_NAME],
```

Comment:
Are you really locking the tooltip to a feature or a location? What about locations with multiple features?

I think `setTooltipState` should take an array of features and track the index of the feature driving the tooltip. Then users can cycle through all of the features when they overlap and get a nice title saying "1 of 10" with next and back arrows.

This would lock the tooltip to a location and the features and not one feature.

Reply:
agreed, but out of scope here imho. I would create separate PR for https://github.com/elastic/kibana/issues/29005, which will warrant a modification of the tooltip-state. but not required now.